autoprecompiles/src/lib.rs

```diff
@@ -326,11 +328,22 @@ fn add_guards<T: FieldElement>(
         id: max_id,
     });
 
-    machine.constraints = machine
-        .constraints
-        .into_iter()
-        .map(|c| add_guards_constraint(c.expr, &is_valid).into())
-        .collect();
+    machine.constraints = if strict {
+        machine
+            .constraints
+            .into_iter()
+            .map(|mut c| {
+                c.expr = is_valid.clone() * c.expr.clone();
+                c
+            })
+            .collect()
+    } else {
```

Comment:
Does this disable some cheaper stuff that `add_guards_constraint` does in the non strict case? It seems like `is_strict` could be pulled into `add_guards_constraint`